code/modules/food/kitchen/smartfridge.dm

```diff
@@ -239,10 +70,42 @@
 		update_icon()
 
 /obj/machinery/smartfridge/update_icon()
-	if(machine_stat & (BROKEN|NOPOWER))
-		icon_state = icon_off
+	overlays.Cut()
+	if(inoperable())
+		icon_state = "[icon_base]-off"
 	else
-		icon_state = icon_on
+		icon_state = icon_base
+
+	if(is_secure)
+		overlays += image(icon, "[icon_base]-sidepanel")
+
+	if(panel_open)
+		overlays += image(icon, "[icon_base]-panel")
+
+	var/image/I
+	var/is_off = ""
+	if(inoperable())
+		is_off = "-off"
+
+	// Fridge contents
+	switch(length(contents))
+		if(0)
+			I = image(icon, "empty[is_off]")
+		if(1 to 2)
+			I = image(icon, "[icon_contents]-1[is_off]")
+		if(3 to 5)
+			I = image(icon, "[icon_contents]-2[is_off]")
+		if(6 to 8)
+			I = image(icon, "[icon_contents]-3[is_off]")
+		else
+			I = image(icon, "[icon_contents]-4[is_off]")
+	overlays += I
+
+	// Fridge top
+	I = image(icon, "[icon_base]-top")
+	I.pixel_z = 32
+	I.layer = ABOVE_WINDOW_LAYER
+	overlays += I
 
```

Comment:
none of these but that last one (since you're modifying pixel z and layer) require images, just add strings